### PR TITLE
Fix doc of the type having generic superclass

### PR DESCRIPTION
**Pull request by MakeNowJust**
For example, [`Channel::Buffered`'s doc](https://crystal-lang.org/api/0.20.0/Channel/Buffered.html):

I fixed it to:

### CODE REVIEW

**Comment by MakeNowJust** [comment]
#3616 is related problem, I know. Because two problems are occurred by [same lines](https://github.com/crystal-lang/crystal/blob/b8602fbd0ce920fc35c2bf3109c636e2798d751c/src/compiler/crystal/tools/doc/type.cr#L59-L60):

```crystal
    when GenericInstanceType
      type.type_vars
```

It is just a bug because `GenericInstanceType#type_vars` returns no `Array(String)` but `Hash(String, ASTNode)`

I think we should not use `Type#full_name` for instance type because we already have nearly same processes such like [`GenericInstanceType#to_s_with_options`](https://github.com/crystal-lang/crystal/blob/master/src/compiler/crystal/types.cr#L1621) and [`Type#type_to_html` for `GenericInstanceType`](https://github.com/crystal-lang/crystal/blob/master/src/compiler/crystal/tools/doc/type.cr#L620). `Type#full_name` should only use for the title of doc page.

**Comment by asterite** [comment]
@MakeNowJust Looks good! Just make sure that the file is properly formatted :-)

File: src/compiler/crystal/tools/doc/type.cr [hunk 1]
```diff
@@ -618,49 +616,52 @@ class Crystal::Doc::Type
...
-      end
     else
-      if text
-        io << text
-      else
-        generic_type.full_name_without_type_vars(io)
+      generic_type.full_name_without_type_vars(io)
+      if not_instantiated
+        io << '('
+        type.type_vars.values.join(", ") { |type_var| type_var.as(Var).type.to_s io }
```
Comment (Sija) [inline src/compiler/crystal/tools/doc/type.cr]: Could that be shortened to `type.type_vars.values.join ", ", &.as(Var).type.to_s(io)`?

Comment (MakeNowJust) [inline src/compiler/crystal/tools/doc/type.cr]: Thanks! I fixed it.

**Comment by MakeNowJust** [comment]
@asterite Sorry, I missed `crystal tool format`.

**Comment by asterite** [comment]
@MakeNowJust Thank you! 💙

**Pull request merged**